### ISSUE TITLE
The `var-name-mixedcase` rule triggers when using `$` as a variable name

### ISSUE THREAD

**Issue by ernestognw** [protofire/solhint#452]
## Description

Given Solidity allows using a `$` as a valid variable identifier, consider the following example:

```solidity
contract A {
    function example() external pure returns (uint256) {
        uint256 $ = 1;
        return $;
    }
}
```

The variable `$` is marked as `Linter: Variable name must be in mixedCase [var-name-mixedcase]`.

## Proposed solution

The library should not throw an error when the variable includes a `$`

**Comment by dbale-altoros** [comment]
This is solved in this PR
https://github.com/protofire/solhint/pull/465
